crates/gitbutler-tauri/src/tests/mod.rs

```diff
@@ -0,0 +1,3 @@
+#[path = "../../../gitbutler-core/tests/shared/mod.rs"]
```

Comment:
I think the culprit is this hack that I introduced to allow code-sharing, without having to put the test code into its own crate which would have been the right thing to do.

To resolve the warnings, the shared test-setup code would have to become its own crate, `gitbutler-testtools` maybe (but I never come up with good names).

PS: I did try to put a hack on the hack by using `pub mod shared;` instead, but Rust is smart enough to know that nothing is truly exported from a binary.

Reply:
Since this is a large PR, one could consider to follow it up with the proper fix as well, and just silence the warnings for now. As I introduced the issue, I'd offer to wrangle it down as well :D.

Reply:
The tests under `gitbutler-core/` still have `pub mod shared` unless I'm looking in the wrong spot. Would this not be fixed just by having these tests as part of the crate like they were before?

Reply:
> Since this is a large PR, one could consider to follow it up with the proper fix as well, and just silence the warnings for now. 

Sure, I can do that :)

Reply:
> The tests under `gitbutler-core/` still have `pub mod shared` unless I'm looking in the wrong spot. Would this not be fixed just by having these tests as part of the crate like they were before?

Do you mean to turn them back into unit-tests and put them under `src/`. I think figuring out what parts need sharing, and making them shareable via a separate crate is the main difficulty that thus far I skipped over.

> Sure, I can do that :)

Ok, please feel free to add a `TODO(ST)` there near the suppression as well.